Core/Utilities/ElasticSearchDB.py

```diff
@@ -0,0 +1,279 @@
+    
+    :param self: self reference
+         
+    """
+    try:
+      if self.__client.ping():
+        result = self.__client.info()
+        self.setClusterName ( result.get( "cluster_name", " " ) )
+        self.log.info( "Database info", result )
+        self._connected = True
+      else:
+        self._connected = False
+        self.log.error( "Cannot connect to the database!" )
+    except ConnectionError as e:
+      self.log.error( e )
+      self._connected = False 
+
+  ########################################################################
+  def getIndexes( self ):
+    """
+    It returns the available indexes...
+    """
+    return [ index for index in self.__client.indices.get_aliases() ]
+  
+  ########################################################################
+  def getDocTypes( self, indexName ):
+    """
+    :param str indexName is the number of the index...
+    :return S_OK or S_ERROR 
+    """
+    result = []
+    try:
+      result = self.__client.indices.get_mapping( indexName )
+    except Exception as e:
+      gLogger.error( e )
+    doctype = ''
+    for i in result:
+      if not result[i].get( 'mappings' ) :
+        return S_ERROR( "%s does not exists!" % indexName )
+      doctype = result[i]['mappings']
+      break
+    return S_OK( doctype ) 
+  
+  ########################################################################
+  def isExists( self, indexName ):
+    """
+    it checks the existance of an index
+    :param str indexName: the name of the index
+    """
+    return self.__client.indices.exists( indexName )
+  
+  ########################################################################
+  def _generateFullIndexName( self, indexName ):
+    """
+    Given an index perfix we create the actual index name.
+    :param str indexName: it is the name of the index
+    """
+    today = datetime.today().strftime( "%Y-%m-%d" )
+    return "%s-%s" % ( indexName, today )
+  
+  def createIndex( self, indexPrefix, mapping ):
+    """
+    :param str indexPrefix: it is the index name. 
+    :param dict mapping: the configuration of the index.
+    
+    """
+    result = None
+    fullIndex = self._generateFullIndexName( indexPrefix )  # we have to create the an index in each day...
+    if self.isExists( fullIndex ):
+      result = S_OK( fullIndex )
+    else:
+      try:
+        self.log.info( "Create index: ", fullIndex + str( mapping ) )
+        self.__client.indices.create( fullIndex, body = mapping )
+        result = S_OK( fullIndex )
+      except Exception as e:
+        self.log.error( "Can not create the index:", e )
+        result = S_ERROR( e )
+    return result
+  
+  def deleteIndex( self, indexName ):
+    """
+    :param str indexName the name of the index to be deleted...
+    """
+    try:
+      retVal = self.__client.indices.delete( indexName )
+    except NotFoundError as e:
+      return S_ERROR(e)
```

Comment:
well either you create a new error code, or you can reuse one which has the same semantic